retrievalmarket/impl/requestvalidation/requestvalidation.go

```diff
@@ -106,9 +106,10 @@ func (rv *ProviderRequestValidator) validatePull(receiver peer.ID, proposal *ret
 	}
 
 	pds := retrievalmarket.ProviderDealState{
-		DealProposal:   *proposal,
-		Receiver:       receiver,
-		LegacyProtocol: legacyProtocol,
+		DealProposal:    *proposal,
+		Receiver:        receiver,
+		LegacyProtocol:  legacyProtocol,
+		CurrentInterval: proposal.PaymentInterval,
```

Comment:
Not setting the `CurrentInterval ` here creates issues when we have a non zero price for Unsealing & bytes because we don't initialize this value correctly in the FSM when we jump directly to the `Give me money for the unsealing` state via the `RequestPayment` event instead of going via the `ProviderOpen` event .